File: android/edge-frameworks-core/src/main/kotlin/io/github/rajibsarwar/edgeframeworks/EdgeRAGOrchestrationModels.kt
package io.github.rajibsarwar.edgeframeworks

data class EdgeRAGRequest(
    val query: String,
    val collection: EdgeKnowledgeCollection? = null,
    val filter: EdgeVectorFilter? = null,
    val topK: Int = 3,
    val minimumScore: Float? = null,
    val retrievalMode: EdgeRetrievalMode =
        EdgeRetrievalMode.VECTOR,
    val systemPrompt: String? = null
)

data class EdgeRAGResult(
    val answer: String,
    val retrievedResults: List<EdgeSearchResult>,
    val context: String,
    val metrics: EdgeRAGMetrics? = null
)

File: ios/Sources/EdgeFrameworks/Core/EdgeRAGOrchestrationModels.swift
public struct EdgeRAGRequest: Sendable {
    public let query: String
    public let collection: EdgeKnowledgeCollection?
    public let filter: EdgeVectorFilter?
    public let topK: Int
    public let minimumScore: Float?
    public let retrievalMode: EdgeRetrievalMode
    public let systemPrompt: String?

    public init(
        query: String,
        collection: EdgeKnowledgeCollection? = nil,
        filter: EdgeVectorFilter? = nil,
        topK: Int = 3,
        minimumScore: Float? = nil,
        retrievalMode: EdgeRetrievalMode = .vector,
        systemPrompt: String? = nil
    ) {
        self.query = query
        self.collection = collection
        self.filter = filter
        self.topK = topK
        self.minimumScore = minimumScore
        self.retrievalMode = retrievalMode
        self.systemPrompt = systemPrompt
    }
}

public struct EdgeRAGResult: Sendable, Equatable {
    public let answer: String
    public let retrievedResults: [EdgeSearchResult]
    public let context: String
    public let metrics: EdgeRAGMetrics?

    public init(
        answer: String,
        retrievedResults: [EdgeSearchResult],
        context: String,
        metrics: EdgeRAGMetrics? = nil
    ) {
        self.answer = answer
        self.retrievedResults = retrievedResults
        self.context = context
        self.metrics = metrics
    }
}
